### PR TITLE
Add clouds.yaml support

### LINKED ISSUE #2: Implement clouds.yaml handling
Currently both `openstack` and `huaweicloud` providers support using `clouds.yaml` file for cloud configuration. On the other side `opentelekomcloud` provider does not

### PR DESCRIPTION
**Pull request by outcatcher**
Resolve #2 

This change inspired with code from terraform-provider-huaweicloud/commit/ab4031cd68db790d845f1bc990675780a9767a32 adapted to opentelekomcloud provider

@anton-sidelnikov Please check implementation details
@gtema Please check if it's compliant with openstack `clouds.yaml` approach

### CODE REVIEW

**Comment by outcatcher** [comment]
Please don't merge yet, acceptance test was not executed

File: opentelekomcloud/config_test.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      project_name: {{.TenantName}}
+      username: {{.Username}}
+      password: {{.Password}}
+      domain_name: {{.DomainName}}
+    region_name: {{.Region}}
+    verify: {{not .Insecure}}
+    cert: {{.ClientCertFile}}
+    key: {{.ClientKeyFile}}
+    cacert: {{.CACertFile}}
+`)
```
Comment (anton-sidelnikov) [inline opentelekomcloud/config_test.go]: is this right indent?

Comment (outcatcher) [reply opentelekomcloud/config_test.go]: Yes

File: opentelekomcloud/config.go [hunk 2]
```diff
@@ -99,6 +112,40 @@ func (c *Config) LoadAndValidate() error {
...
+	// Auth data
+	c.TenantName = ao.TenantName
+	c.TenantID = ao.TenantID
+	c.DomainName = ao.DomainName
+	c.DomainID = ao.DomainID
+	c.IdentityEndpoint = ao.IdentityEndpoint
+	c.Token = ao.TokenID
+	c.Username = ao.Username
+	c.UserID = ao.UserID
+	c.Password = ao.Password
```
Comment (gtema) [inline opentelekomcloud/config.go]: Can we test this all? clouds.yaml contain much more things that those (i.e. user_domain_name, project_id, etc - see https://github.com/terraform-providers/terraform-provider-openstack/blob/master/openstack/provider.go)

Comment (outcatcher) [reply opentelekomcloud/config.go]: I will add new configuration tests for `clouds.yaml`.
OpenTelekomCloud provider configuration doesn't contain project/user arguments (e.g. `ProjectDomainName`/`UserDomainName`). So I will implement it as described in https://github.com/OpenTelekomCloud/terraform-provider-opentelekomcloud/blob/master/website/docs/index.html.markdown#configuration-reference (mapping different vars to single configuration value by priority). Will it suffice?
Proper implementation of project/user arguments is out of the scope of current PR

Comment (outcatcher) [reply opentelekomcloud/config.go]: `clouds.yaml` parsing and processing is handled by `gophercloud/utils`

I've added domain_id/domain_name and project_id/project_name tests (and found differences with documented in `opentelekomcloud` provider specification for `OS_` vars usage)

Also, due to provider specification, 
```markdown
The `region`, `tenant_id`, `domain_id`, `user_id` arguments has been deprecated and `tenant_name`, `domain_name` changed to be `required`
````

**Comment by outcatcher** [comment]
After review, https://github.com/terraform-providers/terraform-provider-opentelekomcloud/pull/434 created

**Pull request closed without merging**